Library/Homebrew/formula.rb

```diff
@@ -105,12 +107,22 @@ def initialize(name, path, spec)
     set_spec :head
 
     @active_spec = determine_active_spec(spec)
+    @active_spec_sym = if head? then :head elsif devel? then :devel else :stable end
     validate_attributes!
-    @pkg_version = PkgVersion.new(version, revision)
     @build = active_spec.build
     @pin = FormulaPin.new(self)
   end
 
+  # @private
+  def set_active_spec(requested)
```

Comment:
This method is used to switch spec between `:stable`, `:devel` and `:head`. What better name do you have in mind.